frontend/packages/enduser-frontend/src/store/modules/forms.ts

```diff
@@ -221,9 +222,29 @@ const module: Module<any, RootState> = {
     removeChildren({ commit }, index) {
       commit(types.REMOVE_CHILDREN, index)
     },
-    async updateUrgentFiles({ commit }, { file, id }) {
-      const { data } = await applicationApi.saveAttachment(id, file)
-      commit(types.UPDATE_URGENT_FILES, { id: data, file })
+    async addUrgentFile({ commit }, { file, applicationId }) {
+      const key = getUUID()
+      commit(types.ADD_URGENT_FILE, { key, file, progress: 0 })
+      const onUploadProgress = (event) => {
+        const progress = Math.round((event.loaded / event.total) * 100)
+        commit(types.UPDATE_URGENT_FILE, { key, file, progress })
+      }
+      const { data } = await applicationApi.saveAttachment(
+        applicationId,
+        file,
+        onUploadProgress
+      )
+      commit(types.UPDATE_URGENT_FILE, {
+        key,
+        id: data,
+        file,
+        progress: 100,
+        done: true
+      })
+    },
```

Comment:
Nothing will catch any upload errors. For example, I just hit 413 Request Entity Too Large when testing locally and it got stuck at:
![image](https://user-images.githubusercontent.com/5830147/99664878-879fcd00-2a71-11eb-852c-82b1c60b5a71.png)

(and obviously threw an error but that isn't visible to any normal users).

Reply:
Yes, I'm aware. Since the error handling is a bit more complicated than usually, it's going to be done later.

Reply:
OK, didn't see the task in Miro

Reply:
Would've been nice to have that in the summary tough :P